crates/sqlbuiltins/src/functions/table/system/cache_external_tables.rs

```diff
@@ -0,0 +1,294 @@
+use crate::functions::table::TableFunc;
+use crate::functions::ConstBuiltinFunction;
+use async_trait::async_trait;
+use datafusion::arrow::array::{StringBuilder, UInt32Builder};
+use datafusion::arrow::datatypes::Schema;
+use datafusion::arrow::record_batch::RecordBatch;
+use datafusion::datasource::TableProvider;
+use datafusion::error::{DataFusionError, Result as DataFusionResult};
+use datafusion::execution::TaskContext;
+use datafusion::physical_expr::PhysicalSortExpr;
+use datafusion::physical_plan::stream::RecordBatchStreamAdapter;
+use datafusion::physical_plan::{
+    DisplayAs, DisplayFormatType, ExecutionPlan, Partitioning, SendableRecordBatchStream,
+    Statistics,
+};
+use datafusion_ext::errors::{ExtensionError, Result};
+use datafusion_ext::functions::{FuncParamValue, TableFuncContextProvider, VirtualLister};
+use datafusion_ext::system::SystemOperation;
+use datasources::native::access::{NativeTableStorage, SaveMode};
+use futures::{stream, Future, StreamExt};
+use protogen::metastore::types::catalog::FunctionType;
+use protogen::metastore::types::catalog::{CatalogEntry, RuntimePreference, TableEntry};
+use std::any::Any;
+use std::collections::HashMap;
+use std::fmt;
+use std::pin::Pin;
+use std::sync::Arc;
+use tracing::warn;
+
+use super::SystemOperationTableProvider;
+use crate::builtins::GLARE_CACHED_EXTERNAL_DATABASE_TABLES;
+use crate::functions::table::virtual_listing::get_virtual_lister_for_external_db;
+
+#[derive(Debug, Clone, Copy)]
+pub struct CacheExternalDatabaseTables;
+
+impl ConstBuiltinFunction for CacheExternalDatabaseTables {
+    const NAME: &'static str = "cache_external_database_tables";
+    const DESCRIPTION: &'static str = "Cache tables from external databases.";
+    const EXAMPLE: &'static str = "select * from cache_external_database_tables();";
```

Comment:
Something for another day, but it'd be cool to support the `CALL` syntax for this.


while valid, `select * from cache_external_database_tables();` seems weird to me as im selecting from an action.

`CALL` seems fitting for something that is just executing a function call. 

```sql
CALL cache_external_database_tables();`
```

Reply:
I agree, I originally wanted to use CALL, but sqlparser doesn't support it.